content/docs/refs-and-the-dom.md

```diff
@@ -11,33 +11,34 @@ redirect_from:
 permalink: docs/refs-and-the-dom.html
 ---
 
-Refs provide a way to access DOM nodes or React elements created in the render method.
+Refs fornecem uma forma de acessar os nós do DOM ou elementos React criados no método render.
 
-In the typical React dataflow, [props](/docs/components-and-props.html) are the only way that parent components interact with their children. To modify a child, you re-render it with new props. However, there are a few cases where you need to imperatively modify a child outside of the typical dataflow. The child to be modified could be an instance of a React component, or it could be a DOM element. For both of these cases, React provides an escape hatch.
+Em um fluxo de dados típico do React, as [props](/docs/components-and-props.html) são a única forma de componentes pais interagirem com seus filhos. Para modificar um componente filho, você terá que re-renderizá-lo com as novas props. Porém, existem alguns casos onde você precisa modificar imperativamente um componente filho fora do fluxo típico de dados. O componente filho a ser modificado poderia ser uma instância de um componente React, ou ele poderia ser um elemento DOM. Para ambos os casos, o React fornece uma saída.
 
-### When to Use Refs {#when-to-use-refs}
+### Quando usar Refs {#quando-usar-refs}
 
-There are a few good use cases for refs:
+Existem alguns bons casos de uso para refs:
 
-* Managing focus, text selection, or media playback.
-* Triggering imperative animations.
-* Integrating with third-party DOM libraries.
+- Geranciamento de foco, seleção de texto, ou reprodução de mídia.
+- Engatilhar animações imperativas.
+- Integração com bibliotecas DOM de terceiros.
 
-Avoid using refs for anything that can be done declaratively.
+Evite usar refs para qualquer coisa que possa ser feita de forma declarativa.
 
-For example, instead of exposing `open()` and `close()` methods on a `Dialog` component, pass an `isOpen` prop to it.
+Por exemplo, ao invés de expôr os métodos `open()` e `close()` em um componente `Dialog`, passe a propriedade `isOpen` para ele.
 
-### Don't Overuse Refs {#dont-overuse-refs}
+### Não utilize Refs excessivamente {#nao-utilize-refs-excessivamente}
```

Comment:
we changed our mind and decided not to translate link sources. could you please change it back?